tests/integration/precompiles/erc20factory/test_types.go

```diff
@@ -0,0 +1,224 @@
+package erc20factory
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/cosmos/evm/precompiles/erc20factory"
+	utiltx "github.com/cosmos/evm/testutil/tx"
+)
+
+func (s *PrecompileTestSuite) TestParseCalculateAddressArgs() {
+	s.SetupTest()
+
+	testcases := []struct {
+		name        string
+		args        []interface{}
+		expPass     bool
+		errContains string
+	}{
+		{
+			name: "pass - correct arguments",
+			args: []interface{}{
+				uint8(0),
+				[32]uint8{},
+			},
+			expPass: true,
+		},
+		{
+			name: "fail - invalid tokenType",
+			args: []interface{}{
+				"invalid tokenType",
+				[32]uint8{},
+			},
+			errContains: "invalid tokenType",
+		},
+		{
+			name: "fail - invalid salt",
+			args: []interface{}{
+				uint8(0),
+				"invalid salt",
+			},
+		},
+		{
+			name: "fail - invalid number of arguments",
+			args: []interface{}{
+				1, 2, 3,
+			},
+			errContains: "invalid number of arguments",
+		},
+	}
+
+	for _, tc := range testcases {
+		s.Run(tc.name, func() {
+			tokenType, salt, err := erc20factory.ParseCalculateAddressArgs(tc.args)
+			if tc.expPass {
+				s.Require().NoError(err, "unexpected error parsing the calculate address arguments")
+				s.Require().Equal(tokenType, tc.args[0], "expected different token type")
+				s.Require().Equal(salt, tc.args[1], "expected different salt")
+			} else {
+				s.Require().Error(err, "expected an error parsing the calculate address arguments")
+				s.Require().ErrorContains(err, tc.errContains, "expected different error message")
+			}
+		})
+	}
+}
+
+func (s *PrecompileTestSuite) TestParseCreateArgs() {
+	addr := utiltx.GenerateAddress()
+	decimals := uint8(18)
+	amount := big.NewInt(1000000)
+	name := "Test"
+	symbol := "TEST"
+
+	s.SetupTest()
+
+	testcases := []struct {
+		name        string
+		args        []interface{}
+		expPass     bool
+		errContains string
+	}{
+		{
+			name: "pass - correct arguments",
+			args: []interface{}{
+				uint8(0),
+				[32]uint8{},
+				name,
+				symbol,
+				decimals,
+				addr,
+				amount,
+			},
+			expPass: true,
+		},
+		{
+			name: "fail - invalid tokenType",
+			args: []interface{}{
+				"invalid tokenType",
+				[32]uint8{},
+				name,
+				symbol,
+				decimals,
+				addr,
+				big.NewInt(1000000),
+			},
+		},
+		{
+			name: "fail - invalid salt",
+			args: []interface{}{
+				uint8(0),
+				"invalid salt",
+				name,
+				symbol,
+				decimals,
+				addr,
+				big.NewInt(1000000),
+			},
+		},
+		{
+			name: "fail - invalid name",
+			args: []interface{}{
+				uint8(0),
+				[32]uint8{},
+				uint8(0),
+				symbol,
+				decimals,
+				addr,
+				big.NewInt(1000000),
+			},
+			errContains: "invalid name",
+		},
+		{
+			name: "fail - invalid symbol",
+			args: []interface{}{
+				uint8(0),
+				[32]uint8{},
+				name,
+				"is",
```

Comment:
This should probably be a valid symbol, nothing in the bank metadata validation prevents symbols of under 3 characters to be added.